devlib/collector/perfetto.py

```diff
@@ -0,0 +1,72 @@
+#    Copyright 2023 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import time
+import subprocess
+from shlex import quote
+
+from devlib.collector import (CollectorBase, CollectorOutput,
+                              CollectorOutputEntry)
+from devlib.exception import TargetStableError, HostError
+from devlib.utils.misc import check_output, which, memoized
+
+OUTPUT_PERFETTO_TRACE = 'trace.perfetto-trace'
+
+class PerfettoCollector(CollectorBase):
```

Comment:
Can we add a doc string to give a quick overview of what this collector adds and perhaps links to the getting started guidance?
Also do you know what version of Android added support for this as that would also be useful to know?

Reply:
Sure, added. The Android framework support was added in Android 9 but it's disabled in 9 and 10 by default on non-Pixel phones. I added a check in the init to manually enable it on those two versions. I think it's pretty unlikely anyone would be using this with Android 8 and below at this point?

Reply:
no problem, sure